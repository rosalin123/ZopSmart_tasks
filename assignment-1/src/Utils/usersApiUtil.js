const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url, { method: 'GET' });
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {

    let url = `${baseUrl}/users/${id}`;
    let user = await fetch(url, { method: 'GET' });
    let userJson = await user.json();
    return userJson;

};
