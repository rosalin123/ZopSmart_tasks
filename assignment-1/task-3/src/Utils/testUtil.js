import React from 'react';

export const findbyTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
