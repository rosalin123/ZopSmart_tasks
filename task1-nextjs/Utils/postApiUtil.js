import axios from 'axios';
import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchPosts = async () => {
  let url = `${baseUrl}/posts`;
  let posts = await axios.get(url);
  let postsJson = await posts.json();
  return postsJson;
};

export const getPost = async (id) => {
  let url = `${baseUrl}/posts/${id}`;
  let post = fetch(url);
  let postJson = await post.json();
  return postJson;
};

export const getPostComments = async (id) => {
  let url = `${baseUrl}/posts/${id}/comments`;
  let comments = await axios.get(url);
  let commentsJson = await comments.json();
  return commentsJson;
};

export const getUserPosts = async (id) => {
  let url = `${baseUrl}/posts?userId=${id}`;
  let userPosts = await fetch(url);
  let userPostsJson = await userPosts.json();
  return userPostsJson;
};
