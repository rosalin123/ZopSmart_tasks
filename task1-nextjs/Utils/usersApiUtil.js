import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url);
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {
  let url = `${baseUrl}/users/${id}`;
  let user = await fetch(url);
  let userJson = await user.json();
  return userJson;
};
