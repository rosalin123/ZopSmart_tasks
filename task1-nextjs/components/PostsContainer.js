import React from 'react';
import { Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PostComponent from './PostComponent';
import PropTypes from 'prop-types';
import { loadGetInitialProps } from 'next/dist/next-server/lib/utils';

const userStyles = makeStyles({
  root: {
    background: '#eceff1',
  },
  headerStyles: {
    backgroundImage: 'linear-gradient(to bottom,black, lightgrey)',
    width: 800,
    height: 300,
    fontSize: '100px',
    fontWeight: 'bold',
    color: 'white',
  },
});

const PostsContainer = (props) => {
  const classes = userStyles();

  return (
    <Grid
      container
      spacing={3}
      direction="column"
      justify="center"
      alignItems="center"
      className={classes.root}
      data-test="postContainerComponent"
    >
      <Grid item data-test="header">
        <div className={classes.headerStyles}>Posts</div>
      </Grid>
      {props.posts.map((post, index) => {
        return (
          <Grid item key={index} data-test="post">
            <PostComponent
              title={post.title}
              user={props.users
                .filter((user) => user.id === post.userId)
                .reduce((userObj, user) => {
                  userObj = { ...user };
                  return userObj;
                }, {})}
              body={post.body}
              id={post.id}
            />
          </Grid>
        );
      })}
    </Grid>
  );
};

export default PostsContainer;
