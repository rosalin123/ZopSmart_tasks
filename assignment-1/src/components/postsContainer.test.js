import React from 'react';
import { shallow } from 'enzyme';

import PostsContainer from './postsContainer';
import { checkProps, findbyTestAttr } from '../Utils/testUtil';

const setUp = (props) => {
  const component = shallow(<PostsContainer {...props} />);
  return component;
};

describe('Posts container Component', () => {
  describe('Checking prop types', () => {
    it('should not return a warning', () => {
      const expectedProps = {
        posts: ['post1', 'post2'],
        users: ['user1', 'user2'],
      };

      const propsErr = checkProps(PostsContainer, expectedProps);
      expect(propsErr).toBeUndefined();
    });
  });

  describe('with props', () => {
    let component;
    beforeEach(() => {
      const props = { posts: ['post1', 'post2'], users: ['user1', 'user2'] };
      component = setUp(props);
    });

    it('Posts Container should render without throwing an error', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render header', () => {
      let wrapper = findbyTestAttr(component, 'header');
      expect(wrapper.length).toBe(1);
    });

    it('should render posts', () => {
      let wrapper = findbyTestAttr(component, 'post');
      expect(wrapper.length).toBe(2);
    });
  });
});
