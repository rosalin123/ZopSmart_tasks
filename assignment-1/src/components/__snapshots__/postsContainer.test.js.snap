// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts container Component with props Posts Container should render without throwing an error 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-root-1"
  container={true}
  data-test="postContainerComponent"
  direction="column"
  justify="center"
  spacing={3}
>
  <WithStyles(ForwardRef(Grid))
    data-test="header"
    item={true}
  >
    <div
      className="makeStyles-headerStyles-2"
    >
      Posts
    </div>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    data-test="post"
    item={true}
    key="0"
  >
    <PostComponent
      user={
        Object {
          "0": "u",
          "1": "s",
          "2": "e",
          "3": "r",
          "4": "2",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    data-test="post"
    item={true}
    key="1"
  >
    <PostComponent
      user={
        Object {
          "0": "u",
          "1": "s",
          "2": "e",
          "3": "r",
          "4": "2",
        }
      }
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
